### PR TITLE
Add Table of Contents to FAQ Page

### PR DESCRIPTION
**Pull request by kylemeenehan**
Issue:

Reading down through the FAQ document to figure out whether your question is there takes some time.

## What I did

I've added a table of contents to the FAQ page so that it's easier to skim read and check whether the page has what you're looking for.

I've also fixed a typo in "environment"

## How to test

Clicking on the headings links should go to the correct headings.

### CODE REVIEW

**Comment by kylemeenehan** [comment]
I should mention that I've only really tested the links with a markdown preview. Is there guidance for building the docs locally so that I can test with the build result?

There shouldn't be any problems as it's standard formatting, but would be good to double check. :-)

**Comment by jonniebigodes** [comment]
@kylemeenehan, this is amazing thanks for taking the time and helping us improve our documentation. We're thankful for it. Let me check this on my end, and if there's nothing that stands out, I'll be more than glad to merge this. Regarding your previous comment, we have [documentation](https://storybook.js.org/docs/react/contribute/new-snippets) on how to preview the docs locally. 

Hope you have a great day!

Stay safe

**Comment by kylemeenehan** [comment]
> @kylemeenehan, this is amazing thanks for taking the time and helping us improve our documentation. We're thankful for it. Let me check this on my end, and if there's nothing that stands out, I'll be more than glad to merge this. Regarding your previous comment, we have [documentation](https://storybook.js.org/docs/react/contribute/new-snippets) on how to preview the docs locally.
> 
> Hope you have a great day!
> 
> Stay safe

No problem, I'm glad I can help!

I did a quick first pass at those steps and I had some issue building, not sure if this is common?

image

Thanks, you have a great day and stay safe too!

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@kylemeenehan we're almost good to go. There are only a couple of items that require a bit of attention. When you're able, could you please make the changes so that this pull request can be merged?

Regarding your [comment](https://github.com/storybookjs/storybook/pull/19490#issuecomment-1281071912), it seems that indeed there's an issue when you or someone else starts the website in hopes of previewing their work either with a change to the documentation or adding new snippets. I'm going to check the website repository, and if there isn't an issue opened, I'll open it and follow up with the rest of the team for an update and, if required, update the documentation accordingly so that contributors can spin up the website without any issues.

File: docs/faq.md [hunk 1]
```diff
@@ -4,6 +4,37 @@ title: 'Frequently Asked Questions'
 
 Here are some answers to frequently asked questions. If you have a question, you can ask it by opening an issue on the [Storybook Repository](https://github.com/storybookjs/storybook/).
 
+* [How can I opt-out of Angular Ivy?](#how-can-i-opt-out-of-angular-ivy)
+* [How can I opt-out of Angular ngcc?](#how-can-i-opt-out-of-angular-ngcc)
+* [How can I run coverage tests with Create React App and leave out stories?](#how-can-i-run-coverage-tests-with-create-react-app-and-leave-out-stories)
+* [I see `ReferenceError: React is not defined` when using Storybook with Next.js](#i-see-referenceerror-react-is-not-defined-when-using-storybook-with-next.js)
+* [How do I setup Storybook to share Webpack configuration with Next.js?](#how-do-i-setup-storybook-to-share-webpack-configuration-with-next.js)
```
Comment (jonniebigodes) [inline docs/faq.md:11]: @kylemeenehan, these aren't working properly when booting up the website locally, they instead should be:
```md
- [I see `ReferenceError: React is not defined` when using Storybook with Next.js](#i-see-referenceerror-react-is-not-defined-when-using-storybook-with-nextjs)
- [How do I setup Storybook to share Webpack configuration with Next.js?](#how-do-i-setup-storybook-to-share-webpack-configuration-with-nextjs)
```

Comment (kylemeenehan) [reply docs/faq.md]: Thanks for picking that up, and sorry I missed it!

I've made the changes and pushed.

**Review by jonniebigodes — APPROVED** [review]
@kylemeenehan, thanks for taking care of the feedback so promptly. I appreciate it. I've checked, and we're good to go. I'll merge this in a bit. @ndelangen, thanks for the assist on this.

**Pull request merged**